plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/BaseJdbcClient.java

```diff
@@ -978,12 +992,35 @@ protected WriteMapping legacyToWriteMapping(@SuppressWarnings("unused") Connecto
                 .orElseGet(Function::identity);
     }
 
+    protected Function<String, String> tryApplyTopN(Optional<List<SortItem>> sortItems, OptionalLong limit)
+    {
+        if (sortItems.isEmpty() || limit.isEmpty()) {
+            return Function.identity();
+        }
+
+        return orderByFunction()
+                .map(orderByFunction -> (Function<String, String>) sql -> orderByFunction.apply(sql, sortItems.get()))
+                .orElseGet(Function::identity)
```

Comment:
this `orElseGet` is weird. We should not get here at all if `orderByFunction()` i `Optional.empty()`. Right?

Reply:
yes. I'll throw here.